File: 2/Euler2.py
def genFib():
    first = 1
    second = 2
    while True:
        yield second
        first, second = second, first+second

if __name__ == '__main__':
    sum = 0
    for n in genFib():
        if n % 2 == 0:
            sum += n
        if n>= 4000000:
            break
    print sum

File: 002/Euler2.rs
fn main(){
    let mut summation = 0u;

    let mut first = 1u;
    let mut second = 1u;
    while second < 4000000 {
        if second % 2 == 0{summation += second};
        second = first + second;
        first = second - first;
    }
    io::println(fmt!("%u", summation))
}